processor/src/system/mod.rs

```diff
@@ -296,3 +301,60 @@ impl System {
         }
     }
 }
+
+// MEMORY CONTEXT
+// ================================================================================================
+
+/// Represents the ID of a memory context
+#[derive(Clone, Copy, Debug, Default, Eq, Ord, PartialEq, PartialOrd)]
+pub struct ContextId(u32);
+
+impl ContextId {
+    /// Returns the root context ID
+    pub fn root() -> Self {
+        Self(0)
+    }
+
+    /// Returns true if the context ID represents the root context
+    pub fn is_root(&self) -> bool {
+        self.0 == 0
+    }
+}
+
+impl From<u32> for ContextId {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+impl From<ContextId> for u32 {
+    fn from(value: ContextId) -> Self {
+        value.0
+    }
+}
+
+impl From<ContextId> for u64 {
+    fn from(context_id: ContextId) -> Self {
+        context_id.0.into()
+    }
+}
+
+impl From<ContextId> for Felt {
+    fn from(context_id: ContextId) -> Self {
+        u64::from(context_id).into()
+    }
+}
+
+impl Sub for ContextId {
+    type Output = u32;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        self.0 - rhs.0
+    }
+}
```

Comment:
I would get rid of this. As mentioned in one of the previous comments, we shouldn't actually be able to subtract contexts. Or at least, subtracting one context from another does not result in a valid context.

Reply:
See [related comment](https://github.com/0xPolygonMiden/miden-vm/pull/1117#discussion_r1370009826)